[identity] Doc updates for GA (#26787)
Azure/azure-sdk-for-js: sdk/identity/identity/TOKEN_CACHING.md

**Note:** When Azure Identity library credentials are used with Azure service libraries (for example, Azure Blob Storage), the in-memory token caching is active in the `HttpPipeline` layer as well. All `TokenCredential` implementations are supported there, including custom implementations external to the Azure Identity library.

#### Caching cannot be disabled

> As there are many levels of cache it is not possible to clear the cache.

Reviewer: ```suggestion
As there are many levels of cache it's not possible to clear the cache.
```

> As there are many levels of cache it is not possible to clear the cache.

Reviewer: I assume that creating a new credential instance will clear the cache. See https://github.com/Azure/azure-sdk-for-java/pull/36304#discussion_r1293535580. If so, add that detail here too.

Author: done

- Ensures the app can continue to function during an Azure AD outage or disruption.
- Avoids having to prompt users to authenticate each time the process is restarted.

> #### Implementation Note

Reviewer: ```suggestion
#### Implementation note
```

> #### Implementation note

Reviewer: Remove this section, as discussed in the team chat.

Author: done
